staging/src/k8s.io/apiserver/pkg/cel/library/format.go

```diff
@@ -0,0 +1,131 @@
+package library
+
+import (
+	"github.com/google/cel-go/cel"
+	"github.com/google/cel-go/common/types"
+	"github.com/google/cel-go/common/types/ref"
+	apimachineryvalidation "k8s.io/apimachinery/pkg/api/validation"
+	"k8s.io/apimachinery/pkg/util/validation"
+	apiservercel "k8s.io/apiserver/pkg/cel"
+)
+
+// Format provides a CEL library exposing common named Kubernetes string
```

Comment:
Are these our expected usage patterns?

```
// validation expression:
!format.dns1123label.validate(self.metadata.name).hasValue() // check that a field is valid

// validation messageExpression:
format.dns1123label.validate(self.metadata.name).value().join('\n') // construct a message
```

?

Reply:
yeah basically

Reply:
I've been trying this out, and, as far as I can tell, these examples, specifically the messageExpression part can never work. Has this been tested in the real world/have you seen this before?

What I'm seeing is that the cost estimator is looking at the cost, in the message expression of `format.dns1123label.validate(object.metadata.name).value()` and coming out with `cost.Max == 3267`, that's all well and good. However, the input to the join then is a list of strings that it is trying to determine the size of, for `join`. The cost estimator has no idea what the size of this list is, so `sizeEstimate` returns `checker.SizeEstimate{Min: 0, Max: math.MaxUint64}` and the expression cost is blown way out of budget.

Wondering if there's any workaround or fix so that the length of the list input to `join` becomes fixed/capped and, then, would actually allow the `messageExpression` example here to work?